pilot/pkg/model/telemetry.go

```diff
@@ -376,6 +377,25 @@ func (t *Telemetries) Tracing(proxy *Proxy) *TracingConfig {
 // HTTPFilters computes the HttpFilter for a given proxy/class
 func (t *Telemetries) HTTPFilters(proxy *Proxy, class networking.ListenerClass) []*hcm.HttpFilter {
 	if res := t.telemetryFilters(proxy, class, networking.ListenerProtocolHTTP); res != nil {
+		if features.EnableECDSForStats.Get() {
+			filters := res.([]*core.TypedExtensionConfig)
+			result := make([]*hcm.HttpFilter, 0, len(filters))
+			for _, f := range filters {
+				result = append(result, &hcm.HttpFilter{
+					Name: f.Name,
```

Comment:
nit: This could work better with a filter chain as the payload for ECDS instead of multiple filters. But I don't think we have that in xDS yet.

Reply:
before the dynamic config discovery received, would it return HTTP 500?